crates/router/src/connector/cashtocode.rs

```diff
@@ -0,0 +1,421 @@
+        merchant_id: &str,
+    ) -> CustomResult<bool, errors::ConnectorError> {
+        let signature = self
+            .get_webhook_source_verification_signature(request)
+            .change_context(errors::ConnectorError::WebhookSourceVerificationFailed)?;
+        let secret = self
+            .get_webhook_source_verification_merchant_secret(db, merchant_id)
+            .await
+            .change_context(errors::ConnectorError::WebhookSourceVerificationFailed)?;
+        let secret_auth = String::from_utf8(secret.to_vec())
+            .into_report()
+            .change_context(errors::ConnectorError::WebhookSourceVerificationFailed)
+            .attach_printable("Could not convert secret to UTF-8")?;
+        let signature_auth = String::from_utf8(signature.to_vec())
+            .into_report()
+            .change_context(errors::ConnectorError::WebhookSourceVerificationFailed)
+            .attach_printable("Could not convert secret to UTF-8")?;
+        let mut success = false;
+        if signature_auth == secret_auth {
+            success = true;
+        }
+        Ok(success)
+    }
+
+    fn get_webhook_object_reference_id(
+        &self,
+        request: &api::IncomingWebhookRequestDetails<'_>,
+    ) -> CustomResult<api_models::webhooks::ObjectReferenceId, errors::ConnectorError> {
+        let webhook: transformers::CashtocodeObjectId = request
+            .body
+            .parse_struct("CashtocodeObjectId")
+            .change_context(errors::ConnectorError::WebhookReferenceIdNotFound)?;
+
+        Ok(api_models::webhooks::ObjectReferenceId::PaymentId(
+            api_models::payments::PaymentIdType::ConnectorTransactionId(webhook.transaction_id),
+        ))
+    }
+
+    fn get_webhook_event_type(
+        &self,
+        _request: &api::IncomingWebhookRequestDetails<'_>,
+    ) -> CustomResult<api::IncomingWebhookEvent, errors::ConnectorError> {
+        Ok(api::IncomingWebhookEvent::PaymentIntentSuccess)
+    }
+
+    fn get_webhook_resource_object(
+        &self,
+        request: &api::IncomingWebhookRequestDetails<'_>,
+    ) -> CustomResult<serde_json::Value, errors::ConnectorError> {
+        let webhook: transformers::CashtocodeIncomingWebhook = request
+            .body
+            .parse_struct("CashtocodeIncomingWebhook")
+            .change_context(errors::ConnectorError::WebhookReferenceIdNotFound)?;
+        let res_json =
+            utils::Encode::<transformers::CashtocodeIncomingWebhook>::encode_to_value(&webhook)
+                .change_context(errors::ConnectorError::WebhookResourceObjectNotFound)?;
+
+        Ok(res_json)
+    }
+
+    fn get_webhook_api_response(
+        &self,
+        request: &api::IncomingWebhookRequestDetails<'_>,
+    ) -> CustomResult<services::api::ApplicationResponse<serde_json::Value>, errors::ConnectorError>
+    {
+        let status = "EXECUTED".to_string();
+        let id = self
+            .get_webhook_object_reference_id(request)
```

Comment:
Why can't we convert request  into transformers::CashtocodeObjectId and then get id from transaction_id, So that we can avoid checks you added from line 401 to 407